Doc/howto/descriptor.rst

```diff
@@ -1460,13 +1472,25 @@ variables:
     class Type(type):
         'Simulate how the type metaclass adds member objects for slots'
 
-        def __new__(mcls, clsname, bases, mapping):
+        def __new__(mcls, clsname, bases, mapping, **kwargs):
             'Emuluate type_new() in Objects/typeobject.c'
             # type_new() calls PyTypeReady() which calls add_methods()
             slot_names = mapping.get('slot_names', [])
-            for offset, name in enumerate(slot_names):
+            if slot_names:
```

Comment:
I don't like these edits (perhaps just a personal preference).  The existing code is much easier to match to and keep in sync with the C code (which is our objective here).  Also, this pure python equivalent has been tested on advanced python students and doesn't seem to have any issues with intelligibility beyond the intrinsic cyclomatic complexity.

Reply:
I agree.